integrations/nvidia/src/haystack_integrations/components/rankers/nvidia/ranker.py

```diff
@@ -0,0 +1,247 @@
+        self._initialized = False
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize the ranker to a dictionary.
+
+        :return: A dictionary containing the ranker's attributes.
+        """
+        return default_to_dict(
+            self,
+            model=self._model,
+            top_k=self._top_k,
+            truncate=self._truncate,
+            api_url=self._api_url,
+            api_key=self._api_key,
+        )
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "NvidiaRanker":
+        """
+        Deserialize the ranker from a dictionary.
+
+        :param data: A dictionary containing the ranker's attributes.
+        :return: The deserialized ranker.
+        """
+        deserialize_secrets_inplace(data, keys=["api_key"])
+        return default_from_dict(cls, data)
+
+    def warm_up(self):
+        """
+        Initialize the ranker.
+
+        :raises ValueError: If the API key is required for hosted NVIDIA NIMs.
+        """
+        model_kwargs = {}
+        if self._truncate is not None:
+            model_kwargs.update(truncate=self._truncate)
+        self._backend = NimBackend(
+            self._model,
+            api_url=self._api_url,
+            api_key=self._api_key,
+            model_kwargs=model_kwargs,
+        )
+        if not self._model:
+            self._model = _DEFAULT_MODEL
+        self._initialized = True
+
+    @component.output_types(replies=List[Document])
+    def run(
+        self,
+        query: str,
+        documents: List[Document],
+        top_k: Optional[int] = None,
+    ) -> Dict[str, List[Document]]:
+        """
+        Rank a list of documents based on a given query.
+
+        :param query: The query to rank the documents against.
+        :param documents: The list of documents to rank.
+        :param top_k: The number of documents to return.
+        :return: A list of ranked documents
+
+        :raises RuntimeError: If the ranker has not been loaded.
+        :raises TypeError: If the arguments are of the wrong type.
+
+        :return: A dictionary containing the ranked documents.
+        """
+        if not self._initialized:
+            msg = "The ranker has not been loaded. Please call warm_up() before running."
+            raise RuntimeError(msg)
+        if not isinstance(query, str):
+            msg = "Ranker expects the `query` parameter to be a string."
+            raise TypeError(msg)
+        if not isinstance(documents, list):
+            msg = "Ranker expects the `documents` parameter to be a list."
+            raise TypeError(msg)
+        if len(documents) == 0:
+            return {"documents": []}
+        if not all(isinstance(doc, Document) for doc in documents):
+            msg = "Ranker expects the `documents` parameter to be a list of Document objects."
+            raise TypeError(msg)
+        if top_k is not None and (not isinstance(top_k, int) or isinstance(top_k, bool)):
```

Comment:
Why a `bool` ?

Reply:
`isinstance(True/False, int)` -> `True`

Reply:
if we allow True/False for top_k, that's what will happen.

if it's ok to accept True/False as 1/0, i can remove the bool portion of the condition.

imho, a user who pass in a bool is on the path to confusion. i'll defer to your judgement.

Reply:
The `bool` check is not necessary, you're already checking if it's not an `int`. Also the `run` type hint clearly says that it mus be an `int`.

```
In [1]: top_k = 1

In [2]: not isinstance(top_k, int) or isinstance(top_k, bool)
Out[2]: False

In [3]: top_k = True

In [4]: not isinstance(top_k, int) or isinstance(top_k, bool)
Out[4]: True

In [5]: top_k = 0.1

In [6]: not isinstance(top_k, int) or isinstance(top_k, bool)
Out[6]: True

In [7]: top_k = "1"

In [8]: not isinstance(top_k, int) or isinstance(top_k, bool)
Out[8]: True
```